geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java

```diff
@@ -741,8 +741,8 @@ public boolean isPdxTypesRegion() {
         // This is for all regions except pdx Region
         if (!isPdxTypesRegion) {
           // Make sure we are distributing to only those senders whose id
-          // is available on this region
-          if (allGatewaySenderIds.contains(sender.getId())) {
+          // is available on this region and whose state is running
+          if (allGatewaySenderIds.contains(sender.getId()) && sender.isRunning()) {
```

Comment:
can we use the new method hasRunningGatewaySender

Reply:
Done!